src/witty/compile_module.py

```diff
@@ -122,8 +380,17 @@ def compile_module(
         force_rebuild = os.getenv("WITTY_FORCE_REBUILD", "0").lower() in ("1", "true")
 
     module_hash = _generate_hash(
-        source_pyx, source_files, extra_compile_args, extra_link_args, extension_kwargs
+        source, source_files, extra_compile_args, extra_link_args, extension_kwargs
     )
+
+    if source_type == "nanobind":
+        assert name is None, (
+            "For non-PYX builds, the module name should not be given but be part "
+            "of the source file."
+        )
+        name = re.findall(r"NB_MODULE\s*\(\s*(.*),", source)
+        assert len(name) == 1, "More than one NB_MODULE found, only one allowed."
+        name = name[0]
     module_name = name + "_" + module_hash
```

Comment:
```suggestion
    module_name = (name or "") + "_" + module_hash
```

by typing along, it is still possible to get to this line with `name` being None... and that would be an error